### DISCUSSION TITLE
Showing breadcrumbs?

### DISCUSSION THREAD

**Discussion by osa1** [nvim-treesitter/nvim-treesitter#912] [Ideas]
I wonder if it's possible to show breadcrumbs to the cursor location using tree-sitter. For example, when editing Rust code like

```
impl Foo {
    fn bar(&self) {
        fn test() {
            _
        }
    }
}
```

where `_` is the cursor, I'd like to see `Foo > bar > test` somewhere on the screen. (this becomes useful when editing large files or having small screen/ lots of splits etc.)

Is this possible using nvim's tree-sitter support? Is it already done somewhere?

**Comment by theHamsta** [comment]
I think someone created a statusline function that does something like this: `:h nvim_treesitter#statusline()`

**Reply by osa1** [reply]
I don't think there's a help section for that? Where can I learn more about it?

**Reply by stsewd** [reply]
@osa1 there is a help entry  for that  `:h nvim_treesitter#statusline()`, also you can see https://github.com/nvim-treesitter/nvim-treesitter#statusline-indicator

**Reply by trevarj** [reply]
This issue is that the parameter `type_patterns` defaults to `{'class', 'function', 'method'}` and this isn't applicable to all languages.
Rust example:

`:lua print(require('nvim-treesitter').statusline({type_patterns = {'impl',
                    'function'}}))`

**Comment by luisiacc** [comment]
How could I modify the `transform_line` to look like this? `MyClass > my_function`, currently the output on python is something like this:
```
class CustomClass(BaseA, BaseB) -> def some_function(self, arg1, arg2)
```

I would like it like this:
```
CustomClass > some_function
```

current transfrom_fn: 
```lua
transform_fn = function(line) return line:gsub('%s*[%[%(%{]*%s*$', '') end,
```

@stsewd @theHamsta do you guys know any solution to this? I don't know how to modify `'%s*[%[%(%{]*%s*$'` to replace this unwanted things as well.

**Reply by stsewd** [reply]
think we may need to rework the status line plugin, so it captures the name from the identifier instead of relying on regexes

**Reply by luisiacc** [reply]
Cool 👍 this would be a killer feature for treesitter in my opinion.

**Comment by TornaxO7** [comment]
[This module](https://github.com/romgrk/nvim-treesitter-context) is probably that what you are looking for.

**Reply by luisiacc** [reply]
Thanks, this is very useful

**Reply by jinleileiking** [reply]
VERY USEFUL. thanks

**Comment by emseidov** [comment]
The [symbols-outline](https://github.com/simrat39/symbols-outline.nvim) plugin may be useful too.

**Comment by hinell** [comment]
**Take a look at [barbecue.nvim](https://github.com/utilyre/barbecue.nvim)**

https://user-images.githubusercontent.com/91974155/197051920-0e89203e-1f0c-4f3c-af72-e7d5e13340ad.mp4

**Reply by emseidov** [reply]
That's perfect, thanks!

**Comment by jdhao** [comment]
Now we also have [drop.nvim](https://github.com/Bekaboo/dropbar.nvim), which provides the same experience as vs code.
